File: python/binary_search.py
def binary_search(my_list, key):
    size = len(my_list)
    lower = 0
    upper = size -1
    while lower <= upper:
         middle = (lower+upper) / 2
         if key == my_list[middle]:
             return middle
         else:
             if key < my_list[middle]:
                 upper = middle -1
             else:
                 lower = middle +1 
    return -1

list_1 = [20,50,80,90,100]
key = int(input('Enter key:'))
result = binary_search(list_1,key)
if result >= 0:
    print('Result Index :')
    print(result)

else:
    print('Key not found!')

File: c++/data_structures_and_algorithm/binary_search.cpp
#include <iostream>  // Include the header file for input/output operations
using namespace std; // Use the standard namespace to avoid writing std:: before each standard function or object
int binarySearch(int arr[], int n, int target)
{
    int low = 0;
    int high = n - 1;
    while (low <= high)
    {
        int mid = low + (high - low) / 2;
        if (arr[mid] == target)
        {
            return mid;
        }
        if (arr[mid] < target)
        {
            low = mid + 1;
        }
        else
        {
            high = mid - 1;
        }
    }
    return -1;
}
int main()
{
    int arr[] = {2, 4, 6, 8, 10, 12, 14, 16, 18, 20};
    int n = sizeof(arr) / sizeof(arr[0]);
    int target = 10;
    int result = binarySearch(arr, n, target);
    if (result != -1)
    {
        cout << "Element found at index " << result << endl; // If found, print the index
    }
    else
    {
        cout << "Element not found in the array" << endl; // If not found, print a not found message
    }
    return 0; // End of the program, return 0 to indicate successful execution
}
